src/views/List.jsx

```diff
@@ -1,6 +1,26 @@
+import { useState } from 'react';
 import { ListItem } from '../components';
+import Button from '../components/Button';
 
 export function List({ data, listToken }) {
+	const [inputItem, setInputItem] = useState('');
+	// This fuction resets the input field:
+	const handleClear = () => {
+		setInputItem('');
+	};
+	const handleInput = (e) => {
+		setInputItem(e.target.value);
+	};
+	// This helper function matches the user input any part of the item name from the items list
+	const stringMatch = (inputItem, listItem) => {
+		return listItem.toLowerCase().includes(inputItem.toLowerCase());
+	};
+
+	// Applies the helper function to narrow down the data
+	const filterData = data.filter((listItem) =>
+		stringMatch(inputItem, listItem.name),
+	);
+
```

Comment:
I love that you separated the definition of the stringMatch function from the filter method. The code is clean and easy to follow this way.

Reply:
Totally agree! And now it's also reusable should we ever need it!